src/Simulation/Native/src/simulator/wavefunction.hpp

```diff
@@ -90,6 +91,52 @@ class Wavefunction
 
     void flush() const
     {
+        //logic to reorder
+        Fusion fg = fused_.get_fusedgates();
+        auto itemsToFuse = fg.get_items();
+        auto ctrlSet = fg.get_ctrl_set();
+        //getting all qubits to move to lower end of the wfn
+        if (itemsToFuse.size() > 0) {
+            std::vector<unsigned> unionOfAllQubitsInUse;
+            std::unordered_set<unsigned> indicesSet; //set is introduced to guard against duplicate insertion and maintianing original order
```

Comment:
I don't understand the ordering requirement. Suppose, you have two items to fuze, one with set {0, 2} and the other with set {1, 3}. `unionOfAllQubitsInUse` will end up looking like {0, 2, 1, 3} -- what order does this represent?

Reply:
This order represents that quibt {0,2} are acted on first, and then {1,3}. This order helps ensure that the qubits being acted on first occupy the lowest end of the wavefunction, and so on. Although this requirement will be nullified as I make my next optimization change for the reorder routine, for the purpose of this PR it was decided that we maintain original order while reordering

Reply:
What if the sets overlap? {2, 3} and {0, 1, 2} will produce sequence of {2, 3, 0, 1} -- is this expected?
Also, are the indices inside an itemsToFuse[i].get_indices() sorted in some defined way?

Reply:
Yes that is expected! Also I believe the get_indices() is sorted as target first, followed by controls